### ISSUE TITLE
Weird behavior in Conv2d padding when changed to 'reflect'

### ISSUE THREAD

**Issue by Jasonlee1995** [pytorch/pytorch#51711]
![image](https://user-images.githubusercontent.com/49643709/106861754-8ba2c900-6709-11eb-9eb1-d1ff1f4e50e0.png)

When I change the convolution layer change the padding mode to 'reflect', it results weird behavior...

I'm using pytorch 1.7.1 cuda 11.0 stable.

Also asked to pytorch forum but nobody answered.
https://discuss.pytorch.org/t/changing-conv2d-parameters-from-pre-trained-models/110538

**Comment by SsnL** [comment]
it is not supported that you set attributes like `padding` and `padding_mode` after creating the object (rather than as arguments).

**Comment by Jasonlee1995** [comment]
![image](https://user-images.githubusercontent.com/49643709/106973953-33b2a380-6797-11eb-82d5-bc6661fb7def.png)

![image](https://user-images.githubusercontent.com/49643709/106974011-5349cc00-6797-11eb-8fbc-01f492aff9af.png)

I'd do more extra experiments.
- changing .padding, .padding_mode works well (I think?)
- but when combined with dilation, it results weird behavior
